### PR TITLE
Uads 012/backend users

### PR DESCRIPTION
**Pull request by johnchen383**
**Issue:**
No functionality was currently implemented to get, put, post, delete users.

***Note: delete node modules folder and run yarn install to removing dependency warnings***

**Solution:**
Functionality was added to set up backend controllers for get, get by id, post, put and delete.

**Risk:**
None of significance as backend route only in development. Might cause conflict with backend restructuring.

**Reviewed By:**
@serenalau272 @Dan1224444 @danielle161 @viresh-oberoi

### CODE REVIEW

File: src/application/controller/UserController.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  async getUsers(req: Request, res: Response) {
+    const mongoAdapter = MongoAdapter.getInstance();
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+
+    var query = {};
+
+    // const queryParams = req.query;
```
Comment (danielle161) [inline src/application/controller/UserController.ts:17]: Remove commented out code

File: src/application/controller/UserController.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+
+    var id = req.params.id.toString();
+
+    var query = {};
+
+    const queryParams = req.query;
+ /*
```
Comment (danielle161) [inline src/application/controller/UserController.ts]: More commented out code to remove

File: src/application/controller/UserController.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    if (req.query.university){
+      query["university"] = req.query.university; 
+    }
+    
+    //modify gradlevel
+    if (req.query.gradlevel){
+      query["gradLevel.type"] = req.query.gradlevel;
+    }
+
+    //modify club
```
Comment (danielle161) [inline src/application/controller/UserController.ts:114]: More commented out code to remove

**Review by Dan1224444 — CHANGES_REQUESTED** [review]
Please add .vscode to gitignore :)

File: src/application/controller/UserController.ts (hunk 1 above)
Comment (johnchen383) [reply src/application/controller/UserController.ts:17]: possibly useful for developers down the line

File: src/application/controller/UserController.ts (hunk 2 above)
Comment (johnchen383) [reply src/application/controller/UserController.ts]: possibly useful for developers down the line

**Review by viresh-oberoi — COMMENTED** [review]
Not sure if you can see the screenshot I have added, but the asPromise method in MongoAdapter.ts (line 49) still has no definition it can find in the modules. Might have to remove it, and it worked fine when I removed it on my end.

image

**Comment by Dan1224444** [comment]
> 
> 
> Not sure if you can see the screenshot I have added, but the asPromise method in MongoAdapter.ts (line 49) still has no definition it can find in the modules. Might have to remove it, and it worked fine when I removed it on my end.
> 
> image

Can you delete your node_modules folder and package-lock.json and try `yarn install` again and see if that is still happening?

**Comment by viresh-oberoi** [comment]
> Can you delete your node_modules folder and package-lock.json and try `yarn install` again and see if that is still happening?

Yes, it is still happening, and you also suggested in the meeting to potentially remove the asPromise method call, as that seemed to work for the couple of us who were having trouble with it. It seemed like a possible and working solution, so, after discussing with John, I removed it in the latest commit.

**Review by Dan1224444 — APPROVED** [review]
Other than commented code, it should be good

**Pull request merged**